content/en/blog/_posts/Deploying-External-OpenStack-Cloud-Provider-With-Kubeadm.md

```diff
@@ -0,0 +1,742 @@
+This document describes how to install a single-master Kubernetes cluster v1.15 with kubeadm on CentOS, and then deploy an external OpenStack cloud provider and Cinder CSI plugin in order to use Cinder volumes as persistent volumes in Kubernetes.
+
+This cluster will be running on OpenStack VMs so we'll create a few things in OpenStack for it.
+
+* a project/tenant for this kubernetes cluster
+* a user in this project for Kubernetes, to query node information and attach volumes etc
+* a private network and subnet
+* a router for this private network and connect it to a public network for floating IPs
+* a security group for all Kubernetes VMs
+* a VM as master node and a few VMs as worker nodes
+
+The security group will have the following rules to open ports for Kubernetes.
+
+**Master Node**
+
+|Protocol  | Port Number | Description|
+|----------|-------------|------------|
+|TCP |6443|Kubernetes API Server|
+|TCP|2379-2380|etcd server client API|
+|TCP|10250|Kubelet API|
+|TCP|10251|kube-scheduler|
+|TCP|10252|kube-controller-manager|
+|TCP|10255|Read-only Kubelet API|
+
+**Worker Nodes**
+
+|Protocol  | Port Number | Description|
+|----------|-------------|------------|
+|TCP|10250|Kubelet API|
+|TCP|10255|Read-only Kubelet API|
+|TCP|30000-32767|NodePort Services|
+
+**CNI Ports on both master and worker nodes**
+
+|Protocol  | Port Number | Description|
+|----------|-------------|------------|
+|TCP|179|Calico BGP network|
```

Comment:
Calico and Flannel are only referenced here. The example below uses weave. It might be worth removing or calling out that these should only be added if running them as an alternative.